### PR TITLE
[Feature] Add LSVT Data Converter

### PR DESCRIPTION
**Pull request by xinke-wang**
## Motivation

Add data converter for [LSVT](https://rrc.cvc.uab.es/?ch=16)

## Modification

Add
- `tools/data/textdet/lsvt_converter.py`
- `tools/data/textrecog/lsvt_converter.py`

Update
- `docs/en/datasets/det.md`
- `docs/en/datasets/recog.md`

### LSVT

- Step1: Download [train_full_images_0.tar.gz](https://dataset-bj.cdn.bcebos.com/lsvt/train_full_images_0.tar.gz), [train_full_images_1.tar.gz](https://dataset-bj.cdn.bcebos.com/lsvt/train_full_images_1.tar.gz), and [train_full_labels.json](https://dataset-bj.cdn.bcebos.com/lsvt/train_full_labels.json) to `lsvt/`.

  ```bash
  mkdir lsvt && cd lsvt

  # Download LSVT dataset
  wget https://dataset-bj.cdn.bcebos.com/lsvt/train_full_images_0.tar.gz
  wget https://dataset-bj.cdn.bcebos.com/lsvt/train_full_images_1.tar.gz
  wget https://dataset-bj.cdn.bcebos.com/lsvt/train_full_labels.json

  mkdir annotations
  tar -xf train_full_images_0.tar.gz && tar -xf train_full_images_1.tar.gz
  mv train_full_labels.json annotations/ && mv train_full_images_1/*.jpg train_full_images_0/
  mv train_full_images_0 imgs

  rm train_full_images_0.tar.gz && rm train_full_images_1.tar.gz && rm -rf train_full_images_1
  ```

#### Det

- Step2: Generate `instances_training.json` and `instances_val.json` (optional) with the following command:

  ```bash
  # Annotations of LSVT test split is not publicly available, split a validation
  # set by adding --val-ratio 0.2
  python tools/data/textdet/lsvt_converter.py PATH/TO/lsvt
  ```

- After running the above codes, the directory structure should be as follows:

  ```text
  |── lsvt
  │   ├── imgs
  │   ├── instances_training.json
  │   └── instances_val.json (optional)
  ```

#### Recog

- Step2: Generate `instances_training.json` and `instances_val.jsonl` (optional) with the following command:

  ```bash
  # Annotations of LSVT test split is not publicly available, split a validation
  # set by adding --val-ratio 0.2
  # Add --preserve-vertical to preserve vertical texts for training, otherwise
  # vertical images will be filtered and stored in PATH/TO/lsvt/ignores
  python tools/data/textdrecog/lsvt_converter.py PATH/TO/lsvt --nproc 4
  ```

- After running the above codes, the directory structure should be as follows:

  ```text
  ├── lsvt
  │   ├── crops
  │   ├── ignores
  │   ├── train_label.jsonl
  │   ├── val_label.jsonl (optional)
  ```

### CODE REVIEW

File: docs/en/datasets/recog.md [hunk 1]
```diff
@@ -354,3 +355,43 @@ rm IIIT-CVid.zip
...
+
+  mkdir annotations
+  tar -xf train_full_images_0.tar.gz && tar -xf train_full_images_1.tar.gz
+  mv train_full_labels.json annotations/ && mv train_full_images_1/*.jpg train_full_images_0/
+  mv train_full_images_0 imgs
+
+  rm train_full_images_0.tar.gz && rm train_full_images_1.tar.gz && rm -rf train_full_images_1
+  ```
+
+- Step2: Generate `instances_training.json` and `instances_val.jsonl` (optional) with following command:
```
Comment (Mountchicken) [inline docs/en/datasets/recog.md:379]: jsonl

File: tools/data/textrecog/lsvt_converter.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    return idx
+
+
+def main():
+    args = parse_args()
+    root_path = args.root_path
+    print('Processing training set...')
+    num_train_imgs = convert_lsvt(
+        root_path=root_path,
+        split='train',
```
Comment (Mountchicken) [inline tools/data/textrecog/lsvt_converter.py:180]: split = 'training'? I guess.

File: tools/data/textrecog/lsvt_converter.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        ratio=args.val_ratio,
+        preserve_vertical=args.preserve_vertical,
+        format=args.format,
+        nproc=args.nproc)
+    if args.val_ratio > 0:
+        print('Processing validation set...')
+        convert_lsvt(
+            root_path=root_path,
+            split='train',
```
Comment (Mountchicken) [inline tools/data/textrecog/lsvt_converter.py:189]: split='val' ?

File: tools/data/textrecog/lsvt_converter.py [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+                    'illegibility': False
+                }, ...
+            ]
+        }
+    ]
+
+
+    Args:
+        root_path (str): The root path of the dataset
+        split (str): The split of dataset. Namely: training or test
```
Comment (Mountchicken) [inline tools/data/textrecog/lsvt_converter.py:100]: training or val ?

**Comment by xinke-wang** [comment]
Any further comments? @Mountchicken

File: docs/en/datasets/recog.md [hunk 5]
```diff
@@ -354,3 +355,43 @@ rm IIIT-CVid.zip
...
+  mkdir annotations
+  tar -xf train_full_images_0.tar.gz && tar -xf train_full_images_1.tar.gz
+  mv train_full_labels.json annotations/ && mv train_full_images_1/*.jpg train_full_images_0/
+  mv train_full_images_0 imgs
+
+  rm train_full_images_0.tar.gz && rm train_full_images_1.tar.gz && rm -rf train_full_images_1
+  ```
+
+- Step2: Generate `instances_training.jsonl` and `instances_val.jsonl` (optional) with the following command:
+
```
Comment (Mountchicken) [inline docs/en/datasets/recog.md:380]: train_label.jsonl and val_label.jsonl

**Review by Mountchicken — APPROVED** [review]

**Comment by gaotongxiao** [comment]
lint test failing
